donkeycar/parts/datastore_v2.py

```diff
@@ -230,50 +240,66 @@ def __init__(self, base_path, inputs=[], types=[], metadata=[],
         self.catalog_paths = list()
         self.catalog_metadata = dict()
         self.deleted_indexes = set()
+        self._got_updated = False
         has_catalogs = False
 
         if self.manifest_path.exists():
             self.seekeable = Seekable(self.manifest_path, read_only=self.read_only)
             if self.seekeable.has_content():
                 self._read_contents()
             has_catalogs = len(self.catalog_paths) > 0
+
         else:
             created_at = time.time()
             self.manifest_metadata['created_at'] = created_at
             if not self.base_path.exists():
                 self.base_path.mkdir(parents=True, exist_ok=True)
-                print('Created a new datastore at %s' % (self.base_path.as_posix()))
+                print(f'Created a new datastore at {self.base_path.as_posix()}')
             self.seekeable = Seekable(self.manifest_path, read_only=self.read_only)
 
         if not has_catalogs:
             self._write_contents()
             self._add_catalog()
         else:
-            last_known_catalog = os.path.join(self.base_path, self.catalog_paths[-1])
-            print('Using catalog %s' % (last_known_catalog))
-            self.current_catalog = Catalog(last_known_catalog, read_only=self.read_only, start_index=self.current_index)
+            last_known_catalog = os.path.join(self.base_path,
+                                              self.catalog_paths[-1])
+            print(f'Using catalog {last_known_catalog}')
+            self.current_catalog = Catalog(last_known_catalog,
+                                           read_only=self.read_only,
+                                           start_index=self.current_index)
+
+        self.session_id = self.create_session_id()
 
     def write_record(self, record):
-        new_catalog = self.current_index > 0 and (self.current_index % self.max_len) == 0
+        new_catalog = self.current_index > 0 \
+                      and (self.current_index % self.max_len) == 0
         if new_catalog:
             self._add_catalog()
 
         self.current_catalog.write_record(record)
         self.current_index += 1
         # Update metadata to keep track of the last index
         self._update_catalog_metadata(update=True)
+        self._got_updated = True
```

Comment:
We use this flag to track if any records have been written into the tub. Only then we update the session information in the manifest metadata. Otherwise if you'd open an existing Tub, it would always append a new session_id. I changed the name.